Extra specs from mutant

kbrock opened the pull request:
Mutant showed a number of use cases that were not present. This does its best at filling in those gaps.

These were pulled out of #20 to make obvious that these are only here to add coverage, not test new behavior.

Fryguy commented on spec/awesome_spawn_spec.rb:
```diff
@@ -7,10 +7,10 @@
-        allow(subject).to receive(:launch).and_return(["", "", 0])
-        subject.send(run_method, "true", :params => params)
+        allow(subject).to receive(:launch).with("true --user bob", {}).and_return(["", "", 0])
+        subject.send(run_method, "true", params)
```
This actually changes the original spec.  Before it would test the inner hash wasn't modified, now it no longer tests that.

Fryguy commented:
The rest looks good!  Sorry I took so long to review it.

kbrock commented:
@Fryguy Good catch. I added a separate test to check that.
I think mutant was complaining about an edge case that we weren't testing there. So that is why I changed it. But yes, it looks like we were not testing the params changed - and that is more probable
